### ISSUE TITLE
Compile error of klitmus7 generated code

### ISSUE THREAD

**Issue by akiyks** [herd/herdtools7#227]
Hi,

Today's master (db524f69447a52d06939134da0e0dbae2955843b) has regression in klitmus7.
Following litmus test (of perfbook's CodeSamples) ended up in compile error:

```
C C-CCIRIW+o+o+o-o+o-o

{}

P0(int *x)
{
	WRITE_ONCE(*x, 1);
}

P1(int *x)
{
	WRITE_ONCE(*x, 2);
}

P2(int *x)
{
	int r1;
	int r2;

	r1 = READ_ONCE(*x);
	r2 = READ_ONCE(*x);
}

P3(int *x)
{
	int r3;
	int r4;

	r3 = READ_ONCE(*x);
	r4 = READ_ONCE(*x);
}

exists(2:r1=1 /\ 2:r2=2 /\ 3:r3=2 /\ 3:r4=1)
```

Compile error log from "make":

```
make -C /lib/modules/5.4.0-72-generic/build/ M=/home/foo/Lab/klitmus7/klitmus modules
make[1]: Entering directory '/usr/src/linux-headers-5.4.0-72-generic'
  CC [M]  /home/foo/Lab/klitmus7/klitmus/litmus000.o
/home/foo/Lab/klitmus7/klitmus/litmus000.c: In function ‘thread2’:
/home/foo/Lab/klitmus7/klitmus/litmus000.c:484:27: error: ‘ctx_t {aka struct <anonymous>}’ has no member named ‘_a’
       code2(&_a->x[_i],&_a->_a->out_2_r2[_i],&_a->_a->out_2_r1[_i]);
                           ^~
/home/foo/Lab/klitmus7/klitmus/litmus000.c:484:49: error: ‘ctx_t {aka struct <anonymous>}’ has no member named ‘_a’
       code2(&_a->x[_i],&_a->_a->out_2_r2[_i],&_a->_a->out_2_r1[_i]);
                                                 ^~
/home/foo/Lab/klitmus7/klitmus/litmus000.c: In function ‘thread3’:
/home/foo/Lab/klitmus7/klitmus/litmus000.c:518:27: error: ‘ctx_t {aka struct <anonymous>}’ has no member named ‘_a’
       code3(&_a->x[_i],&_a->_a->out_3_r4[_i],&_a->_a->out_3_r3[_i]);
                           ^~
/home/foo/Lab/klitmus7/klitmus/litmus000.c:518:49: error: ‘ctx_t {aka struct <anonymous>}’ has no member named ‘_a’
       code3(&_a->x[_i],&_a->_a->out_3_r4[_i],&_a->_a->out_3_r3[_i]);
                                                 ^~
scripts/Makefile.build:269: recipe for target '/home/foo/Lab/klitmus7/klitmus/litmus000.o' failed
make[2]: *** [/home/foo/Lab/klitmus7/klitmus/litmus000.o] Error 1
Makefile:1754: recipe for target '/home/foo/Lab/klitmus7/klitmus' failed
make[1]: *** [/home/foo/Lab/klitmus7/klitmus] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-5.4.0-72-generic'
Makefile:5: recipe for target 'all' failed
make: *** [all] Error 2
```

It compiles successfully on the previous commit (45690d9d0f7a956a6d3dbaf9e912efb22835756e).

This is on Ubuntu 18.04 with gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0.
uname -r: 5.4.0-72-generic

Can you look into this?

Thanks, Akira

**Comment by maranget** [comment]
Hi @akiyks, the bug should be fixed by PR #228.

Thanks,
--Luc

**Comment by akiyks** [comment]
Hi Luc,

Confirmed the fix. Now I'm closing this issue.
Thank you very much!

--
Akira

> Hi @akiyks, the bug should be fixed by PR #228.
> 
> Thanks,
> --Luc
